pages/tidal tools/vsphere.md

```diff
@@ -6,20 +6,58 @@ sidebar: main_sidebar
 permalink: vsphere.html
 folder: tidaltools
 ---
-## Getting Started
-To get started you will need to make sure you have a copy of [Tidal Tools](https://get.tidal.sh) and it is [installed](tidal-tools.html).
+
+## Overview
+
+As part of your Cloud Migration journey, it is very important to have a reliable inventory of your infrastructure. Tidal Tools is able to connect to a Hypervisor to synchronize your inventory of virtual machines from vSphere.
+
+This guide will walk you through the steps and necessary configuration to properly integrate with vSphere and capture the following data points on its VMs.
+
+```
+- FQDN
+- IP Addresses
+- HostName
+- AssignedID (InstanceUuid)
+- RAM Used (gb)
+- RAMAllocated (gb)
+- CPU Name
+- CPU Count
+- Storage Used (gb)
+- Storage Allocated (gb)
+- Operating System
+- Operating System Version
+- CreatedAt
```

Comment:
does this mean that these dates represent the changes to the VM in vSphere? or is it that these are updated accordingly when the other attributes above on the model are also updated?

asking both in terms of what the design intention is and also in terms of clarity in communicating this in the guide to readers.

Reply:
Thanks for picking up on this @prees1. It was scaping me. 
Yes, those dates represent the changes to the VM in vSphere. Ramiz identified those 2 new data points (CreatedAt, UpdatedAt). I was just forgetting to find out whether or not they are valuable to us and/or they could present a problem.

Should the payload include these 2 fields or we reserve those 2 fields to the API logic

Reply:
lots to consider here!

So the API today, the created_at and updated_at attributes across all resources are used to track and understand changes to the data persistence within the API itself, it is metadata.

It sounds like to me (not familiar with what they are exactly) this created at and updated at values are analogous to the tidal API but in respect to the hypervisor and control plane of the VMs.

- should we include these fields - yea they might be useful for people during migrations so we definitely could. Thinking what they might be useful for, it could be to understand how long or old a VM is, which might be useful for people to filter or track VMs from a certain time period, ie. a hypothetical scenario - 'ignoring ones made after X date as those are temporary. I suppose same with updated at too, maybe useful to know 'oh we updated a configuration change to all the ones that are ready to migrate, so any with a update date after x is good to go'. I would have to know what conditions or scenarios these times represent, to really get a handle on any potential use cases.

- how to persist them - this data is definitely distinct from the data used in our API by the same name. So short of changing that inherent pattern throughout the API, they would be independent of those data points.

Reply:
Keep in mind too there is the workflow of `tidal get vsphere | transform | tidal sync servers`

Reply:
👍 Thanks!

Given where we are, I don't think there is anything else to consider at this moment right? we are not going to use those 2 fields (especially with their names).

Interpreting your suggestion ☝️ correctly,  we could have those 2 fields "transformed" (like renamed) before piping them via sync servers.

Also, we can modify their name to match more closely what they represent. for example,
- vm_created_at
- vm_updated_at
Thinking about the future here. Probably no need to make that change right now. Do you agree?

Reply:
agree. don't think it's on the critical path to the goal right now of machine stats parity.